get.go

```diff
@@ -59,6 +60,12 @@ func Get(jwksURL string, options Options) (jwks *JWKS, err error) {
 // backgroundRefresh is meant to be a separate goroutine that will update the keys in a JWKS over a given interval of
 // time.
 func (j *JWKS) backgroundRefresh() {
+	if j.initAsync {
+		if err := j.refresh(); err != nil && j.refreshErrorHandler != nil {
+			j.refreshErrorHandler(err)
+		}
```

Comment:
This makes no sense to me - the point of this function seems to be running `j.refresh`... why are you running it right here? Was this for testing?

Reply:
Is the issue that if initAsync is set then we want the background refresh to kick off right away rather than waiting for the initial interval?

In this case, it seems you could just write to `j.refreshRequests` after starting the `backgroundRefresh` goroutine in the `Get` function.

Reply:
good idea, I added writing to `backgroundRefresh`